### PR TITLE
refactor: remove unused hook submodule

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

the `hook` submodule is no longer used and can be removed

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: tests/vendor/test_hook.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import mock
 
 from ddtrace.compat import reload_module
-from ddtrace.utils.hook import deregister_post_import_hook
-from ddtrace.utils.hook import register_post_import_hook
+from ddtrace.vendor.hook import deregister_post_import_hook
```
Comment (brettlangdon) [inline tests/vendor/test_hook.py:4]: wait, is this actually used anywhere? if there are only tests for it, delete the entire file?

Comment (P403n1x87) [reply tests/vendor/test_hook.py:4]: There's also `internal/import_hooks.py` that does a similar thing and it's just tested but not used. Is it worth investigating why these two modules are there in the first place?

**Review by jd — CHANGES_REQUESTED** [review]
I'm confused. Vendor is about things we vendor. Did we took hook from somewhere else?

**Comment by P403n1x87** [comment]
> I'm confused. Vendor is about things we vendor. Did we took hook from somewhere else?

It's based off of `wrapt.importer` I think

**Review by majorgreys — APPROVED** [review]

**Review by jd — COMMENTED** [review]
yay to less code

**Pull request merged**